frontend/packages/console-dynamic-plugin-sdk/src/api/api.ts

```diff
@@ -1,11 +1,18 @@
-import { UseK8sWatchResource, UseK8sWatchResources, UseResolvedExtensions } from './api-types';
+import { MockImpl, MockCoFetchJSONImpl } from './api-mocks';
+import {
+  UseK8sWatchResource,
+  UseK8sWatchResources,
+  UseResolvedExtensions,
+  CoFetch,
+  CoFetchJSON,
+  CoFetchText,
+} from './api-types';
 
 export * from './api-types';
 
-const MockImpl = () => {
-  throw new Error('You need to configure webpack externals to use this function at runtime.');
-};
-
 export const useK8sWatchResource: UseK8sWatchResource = MockImpl;
 export const useK8sWatchResources: UseK8sWatchResources = MockImpl;
 export const useResolvedExtensions: UseResolvedExtensions = MockImpl;
+export const coFetch: CoFetch = MockImpl;
```

Comment:
We'll want API doc describing these APIs and how they differ from `fetch`.

@vojtechszocs @christianvogt Have we settled on where APIs are documented? Topic for the next sync meeting?

Reply:
@spadgett As discussed, we can modify `generate-doc.ts` script in dynamic plugin SDK package to parse public APIs (i.e. all exports from ^^ module, possibly traversing other exports transitively) and extract JSDoc from them.

Just like `const getConsoleExtensions: () => ExtensionTypeInfo[]` we have today, we could have `getConsoleAPIs` type info collector, from which we can generate `console-apis.md` document.

Reply:
Added JSDoc comments to `ConsoleFetch`, `ConsoleFetchJSON` and `ConsoleFetchText`.